File: brain.py
from neuron import Neuron
# from ..dataProcessing.brainInput import brainInput
class Brain:
    neurons = [Neuron]
     
    def get_neuron(self, x : int):
        return self.neurons[x]
    
    def createNeuron(self):
        self.neurons.append(Neuron())
        return len(self.neurons)

File: src/brainParts/Brain.cpp
#include "brainParts/Brain.h"

Brain::Brain() { printf("Hello from brain \n"); }

void Brain::Hello() {
  printf("Hello\n");
  neuronMap.insert(std::make_pair<unsigned long, Neuron *>(1, new Neuron()));
}

Neuron *Brain::getNeuron(unsigned long nID) { return neuronMap[nID]; }
Neuron *Brain::createNeuron(unsigned long neuronID) {
  // Neuron *neuronTmp = new Neuron();
  neuronMap.insert(std::make_pair<unsigned long, Neuron *>(neuronID, new Neuron()));
  return neuronMap[1];
}

#ifdef ENABLE_DOCTEST_IN_LIBRARY
#include "doctest.h"
TEST_CASE("we can have tests written here, to test impl. details") {
  CHECK(true);
}
#endif
